fluent-bundle/src/types.ts

```diff
@@ -104,37 +117,134 @@ export class FluentNumber extends FluentType<number> {
+ * as a numerical timestamp in milliseconds. It may also store an
  * option bag of options which will be passed to `Intl.DateTimeFormat` when the
  * `FluentDateTime` is formatted to a string.
  */
-export class FluentDateTime extends FluentType<number> {
+export class FluentDateTime extends FluentType<
+  | number
+  | Date
+  | Temporal.Instant
+  | Temporal.PlainDateTime
+  | Temporal.PlainDate
+  | Temporal.PlainMonthDay
+  | Temporal.PlainTime
+  | Temporal.PlainYearMonth
+  | Temporal.ZonedDateTime
+> {
   /** Options passed to `Intl.DateTimeFormat`. */
   public opts: Intl.DateTimeFormatOptions;
 
+  static supportsValue(value: any): value is ConstructorParameters<typeof Temporal.Instant>[0] {
+    if (typeof value === "number") return true;
+    if (value instanceof Date) return true;
+    if (value instanceof FluentType) return FluentDateTime.supportsValue(value.valueOf());
+    // Temporary workaround to support environments without Temporal
+    if ('Temporal' in globalThis) {
+      if (
+        // @ts-ignore
+        value instanceof Temporal.Instant        || // @ts-ignore
+        value instanceof Temporal.PlainDateTime  || // @ts-ignore
+        value instanceof Temporal.PlainDate      || // @ts-ignore
+        value instanceof Temporal.PlainMonthDay  || // @ts-ignore
+        value instanceof Temporal.PlainTime      || // @ts-ignore
+        value instanceof Temporal.PlainYearMonth || // @ts-ignore
+        value instanceof Temporal.ZonedDateTime
+      ) {
+        return true;
+      }
+    }
+    return false
+  }
+
   /**
    * Create an instance of `FluentDateTime` with options to the
    * `Intl.DateTimeFormat` constructor.
    *
    * @param value The number value of this `FluentDateTime`, in milliseconds.
    * @param opts Options which will be passed to `Intl.DateTimeFormat`.
    */
-  constructor(value: number, opts: Intl.DateTimeFormatOptions = {}) {
+  constructor(
+    value:
+      | number
+      | Date
+      | Temporal.Instant
+      | Temporal.PlainDateTime
+      | Temporal.PlainDate
+      | Temporal.PlainMonthDay
+      | Temporal.PlainTime
+      | Temporal.PlainYearMonth
+      | Temporal.ZonedDateTime
+      | FluentDateTime
+      | FluentType<number>,
+    opts: Intl.DateTimeFormatOptions = {}
+  ) {
+    // unwrap any FluentType value, but only retain the opts from FluentDateTime
+    if (value instanceof FluentDateTime) {
+      opts = { ...value.opts, ...opts };
+      value = value.value;
+    } else if (value instanceof FluentType) {
+      value = value.valueOf();
+    }
+
+    if (typeof value === "object") {
+      // Intl.DateTimeFormat defaults to gregorian calendar, but Temporal defaults to iso8601
+      if ('calendarId' in value) {
+        if (opts.calendar === undefined) {
+          opts = { ...opts, calendar: value.calendarId };
+        } else if (opts.calendar !== value.calendarId && 'withCalendar' in value) {
+          value = value.withCalendar(opts.calendar);
+        }
+      }
```

Comment:
This automatic conversion is probably a bad idea, esp. as the formatter option `calendar` is already excluded from `DATETIME_ALLOWED`.

Reply:
Okay, makes sense. But setting the option is acceptable? I wanted to avoid users having to create their own `FluentDateTime`, and as I said in the comment, `Temporal` defaults to a different calendar than `Intl`.

Reply:
Yeah, if no `calendar` value is set, then it's reasonable to default to the one in the temporal instance.

Reply:
That sounds good (and reasonable). I do wish that this was the `Intl.DateTimeFormat` behavior to begin with.